save-preprocessor/src/main/kotlin/org/cqfn/save/preprocessor/controllers/DownloadProjectController.kt

```diff
@@ -229,11 +232,13 @@ class DownloadProjectController(private val configProperties: ConfigProperties,
         val tmpDir = generateDirectory(binFile.name)
         val pathToProperties = tmpDir.resolve(propertyFile.name)
         propertyFile.copyTo(pathToProperties)
+        propertyFile.delete()
         binFile.copyTo(tmpDir.resolve(binFile.name))
+        binFile.delete()
         val project = executionRequestForStandardSuites.project
         val propertiesRelativePath = pathToProperties.relativeTo(tmpDir).name
         // todo: what's with version?
-        return updateExecution(executionRequestForStandardSuites.project, propertiesRelativePath, binFile.name).flatMap { execution ->
+        return updateExecution(executionRequestForStandardSuites.project, tmpDir.name, binFile.name).flatMap { execution ->
```

Comment:
It still should be a part of path, not only directory name, Ithink

Reply:
This path used like `execution.resourcesRootPath` in `buildBaseImageForExecution()` in this way:
```kotlin
        val resourcesPath = File(
            configProperties.testResources.basePath,
            execution.resourcesRootPath,
        )
```

So we take the `configProperties.testResources.basePath` and then appends the relative path, which is name of directory, so I think all is correct.
Or what did you mean?

Reply:
I actually meant, that there can be several directories, and `name` returns the name of the last one only. But as I see now, in this function we don't really pay attention to test root directory and just copy everything into root. Then, `tmpDir.name` works, but we will need to change general logic later.